src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.PathRewrite.cs

```diff
@@ -0,0 +1,143 @@
+﻿#nullable enable
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Uno.UI.SourceGenerators.XamlGenerator.Utils
+{
+	internal static partial class XBindExpressionParser
+	{
+		private const string XBindSubstitute = "↔↔";
+		private static readonly Regex NamedParam = new Regex(@"^(\w*)=(.*?)");
+
+		/// <summary>
+		/// Rewrites all x:Bind expression Path property to be compatible with the System.Xaml parser.
+		/// </summary>
+		/// <param name="markup">The original markup</param>
+		/// <returns>The adjusted markup</returns>
+		/// <remarks>
+		/// The System.Xaml parser does not support having extended expressions, with
+		/// quotes or commads inside an unquoted property value. The point of this method is
+		/// to encode the whole Path property as Base64 to that its content does not interfere
+		/// with the default rules of the parser. The expression is then restored using <see cref="RestoreSinglePath"/>.
+		/// </remarks>
+		internal static string RewriteDocumentPaths(string markup)
+		{
+			var result =
+				Regex.Replace(
+					markup,
+					 "\"{x:Bind (.*?)}\"",
+					e => $"\"{{x:Bind {RewriteParameters(e.Groups[1].Value)}}}\""
+				);
+
+			return result;
+		}
+
+		/// <summary>
+		/// Restores an x:Bind path encoded with <see cref="RewriteDocumentPaths"/>.
+		/// </summary>
+		internal static string RestoreSinglePath(string path)
+		{
+			if (!string.IsNullOrEmpty(path))
+			{
+				var bytes = Convert.FromBase64String(path.Replace("_", "="));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this call to 'path.Replace' to an overload that accepts a 'StringComparison' as a parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5436550)